emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java

```diff
@@ -169,13 +169,19 @@ public void configureTableJobProperties(TableDesc tableDesc, Map<String, String>
             .getProperty(DynamoDBConstants.THROUGHPUT_WRITE_PERCENT));
       }
 
-      String readThroughput = description.getProvisionedThroughput().getReadCapacityUnits()
-          .toString();
-      String writeThroughput = description.getProvisionedThroughput().getWriteCapacityUnits()
-          .toString();
+      if(description.getBillingModeSummary().getBillingMode().equals(DynamoDBConstants.BILLING_MODE_PROVISIONED)) {
+        jobProperties.put(DynamoDBConstants.READ_THROUGHPUT, description.getProvisionedThroughput().getReadCapacityUnits()
+                .toString());
+        jobProperties.put(DynamoDBConstants.WRITE_THROUGHPUT,description.getProvisionedThroughput().getWriteCapacityUnits()
+                .toString());
+      }
+      else
+      {
+        // If not specified at the table level, set a hard coded value of 40,000
```

Comment:
Why 40k?

Reply:
I would try to get the latest max capacity used, but AFAIK it's not available through the API.
Should the default be another value?